OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -279,6 +290,48 @@ private void initObserver() {
         });
     }
 
+    private void initInAppPurchaseSetup() {
+        if (courseData.getMode().equalsIgnoreCase(EnrollmentMode.AUDIT.toString()) && !isVideoMode) {
+            initFullscreenLoader();
+            initInAppPurchaseObserver();
+        }
+    }
+
+    private void initFullscreenLoader() {
+        // To proceed with the same instance of dialog fragment in case of orientation change
+        this.fullscreenLoader = (FullscreenLoaderDialogFragment) getChildFragmentManager().findFragmentByTag(FullscreenLoaderDialogFragment.TAG);
+        if (this.fullscreenLoader == null) {
+            this.fullscreenLoader = FullscreenLoaderDialogFragment.newInstance();
+        }
+    }
+
+    private void initInAppPurchaseObserver() {
+        iapViewModel = new ViewModelProvider(requireActivity()).get(InAppPurchasesViewModel.class);
+
+        iapViewModel.getShowFullscreenLoaderDialog().observe(getViewLifecycleOwner(), show -> {
+            if (show) {
+                fullscreenLoader.show(getChildFragmentManager(), FullscreenLoaderDialogFragment.TAG);
+                iapViewModel.showFullScreenLoader(false);
+            }
+        });
+
+        iapViewModel.getRefreshCourseData().observe(getViewLifecycleOwner(), refresh -> {
+            if (refresh) {
+                getCourseComponentFromServer(false);
+                iapViewModel.refreshCourseData(false);
+            }
+        });
+
+        iapViewModel.getProcessComplete().observe(getViewLifecycleOwner(), complete -> {
+            if (complete) {
+                courseData.setMode(EnrollmentMode.VERIFIED.toString());
```

Comment:
Still pending

Reply:
Moved it under Success case and call Enrollment API to refresh Course Mode.